docs/overview/transaction.md

```diff
@@ -1,85 +1,131 @@
+in epoch order, regardless of execution order. This baseline model
+delivers the maximum scalability and performance to data models and
+applications that do not generate conflicting I/O workload. Typical
+examples are collective MPI-IO operations, POSIX file read/write
 or HDF5 dataset read/write.
 
-For parts of the data model that require conflict serialization, 
-DAOS provides distributed serializable transaction based on multi-version 
-concurrency control. Transactions are typically needed when different user 
-processes can overwrite the value associated with a dkey/akey pair. 
-Examples are a SQL database over DAOS or a consistent POSIX namespace 
-accessed concurrently by uncoordinated clients. All I/O operations 
-(including reads) submitted in the context of the same operation will use 
-the same epoch. The DAOS transaction mechanism automatically detects the 
-traditional read/write, write/read and write/write conflicts and aborts 
-one of the conflicting transactions (the transaction fails to commit 
-with `-DER_RESTART`). The failed transaction then has to be restarted 
+For parts of the data model that require conflict serialization,
+DAOS provides distributed serializable transaction based on multi-version
+concurrency control. Transactions are typically needed when different user
+processes can overwrite the value associated with a dkey/akey pair.
+Examples are a SQL database over DAOS or a consistent POSIX namespace
+accessed concurrently by uncoordinated clients. All I/O operations
+(including reads) submitted in the context of the same transaction will use
+the same epoch. The DAOS transaction mechanism automatically detects the
+traditional read/write, write/read and write/write conflicts and aborts
+one of the conflicting transactions (the transaction fails to commit
+with `-DER_RESTART`). The failed transaction then has to be restarted
 by the user/application.
 
-In the initial implementation, the transaction API has the following 
-limitations that will be addressed in future DAOS versions:
-
--    no support for the array API
--    transactional object update and key-value put operations are not visible 
-     via object fetch/list and key-value get/list operations executed in the 
-     context of the same transaction.
-
+In the initial implementation, the transaction API does not support reading
+your own uncommitted changes. In other words, transactional object or key-value
+modifications are invisible to the subsequent operations executed in the
+context of the same transaction. The transaction API is supported for all
+object types and can be combined with the the event and scheduler interface.
+
+The typical flow of a transaction is as follows:
+
+```C
+daos_handle_t tx = DAOS_TX_NONE;
+int           rc;
+
+/* allocate transaction */
+rc = daos_tx_open(dfs->coh, &th, 0, NULL);
+if (rc)
+	...
+
+restart:
+	/* execute operations under the same transaction */
+	rc = daos_obj_fetch(..., th);
```

Comment:
fetch also can return restart, but i understand if you want to keep this simple here

Reply:
Yup, let's keep it simple for now ...